File: api/src/Domain/ProductCatalog/DataTransferObjects/PadCreateOrUpdate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using DetailingArsenal.Domain.ProductCatalog;
using DetailingArsenal.Domain.Shared;

namespace DetailingArsenal.Domain.ProductCatalog {
    public class PadCreateOrUpdate : IDataTransferObject {
        public Guid? Id { get; }
        public string Name { get; }
        public List<PadCategory> Category { get; }
        public PadMaterial? Material { get; }
        public PadTexture? Texture { get; }
        public PadColor? Color { get; }
        public bool? HasCenterHole { get; }
        public Either<Guid, DataUrlImage>? Image { get; }
        public List<PadOptionCreateOrUpdate> Options { get; }

        public PadCreateOrUpdate(Guid? id, string name, PadCategory category, PadMaterial? material, PadTexture? texture, PadColor? color, bool? hasCenterHole, Either<Guid, DataUrlImage>? image, List<PadOptionCreateOrUpdate> options) {
            Id = id;
            Name = name;
            Category = new(new[] { category });
            Material = material;
            Texture = texture;
            Color = color;
            HasCenterHole = hasCenterHole;
            Image = image;
            Options = options;
        }

        [JsonConstructor]
        public PadCreateOrUpdate(Guid? id, string name, List<PadCategory> category, PadMaterial? material, PadTexture? texture, PadColor? color, bool? hasCenterHole, Either<Guid, DataUrlImage>? image, List<PadOptionCreateOrUpdate> options) {
            Id = id;
            Name = name;
            Category = category;
            Material = material;
            Texture = texture;
            Color = color;
            HasCenterHole = hasCenterHole;
            Image = image;
            Options = options;
        }
    }
}

File: app/src/api/product-catalog/data-transfer-objects/requests/pad-create-or-update.ts
import { Image } from '../image';
import { PadCategory } from '../pad-category';
import { PadColor } from '../pad-color';
import { PadMaterial } from '../pad-material';
import { PadOption } from '../pad-option';
import { PadTexture } from '../pad-texture';
import { PadOptionCreateOrUpdate } from './pad-option-create-or-update';

export interface PadCreateOrUpdate {
    id: string | null;
    name: string;
    category: PadCategory[];
    material: PadMaterial | null;
    texture: PadTexture | null;
    color: PadColor | null;
    hasCenterHole: boolean | null;
    image: Image | string | null;
    options: PadOptionCreateOrUpdate[];
}
